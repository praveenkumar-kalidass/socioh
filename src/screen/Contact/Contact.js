import React from 'react';

import { CONSTANT } from '../../constant';
import wrapPattern from '../../helper/wrapPattern';

const Contact = () => <></>;

export default wrapPattern({
  Component: Contact,
  pattern: CONSTANT.SCREEN_BACKGROUND.BACKGROUND_2,
});
