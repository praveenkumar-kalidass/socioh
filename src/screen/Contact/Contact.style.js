import styled from 'styled-components/native';
import { Icon, Text } from 'native-base';

import { COLOR, FONT, SPACING } from '../../constant';

export const ContactContainer = styled.View`
  flex: 1;
  justify-content: center;
  padding-horizontal: ${SPACING.SPACE_20};
`;

export const AvatarContainer = styled.View`
  align-items: center;
  flex: 1;
  padding-vertical: ${SPACING.SPACE_20};
`;

export const AvatarLabel = styled(Text)`
  color: ${COLOR.BLACK};
  font-size: ${FONT.SIZE_24};
  margin-top: ${SPACING.SPACE_10};
`;

export const ContactListContainer = styled.View`
  flex: 1;
`;

export const CallIcon = styled(Icon)`
  color: ${COLOR.PRIMARY};
`;
