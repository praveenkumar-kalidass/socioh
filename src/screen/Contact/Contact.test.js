import React from 'react';
import { render } from '@testing-library/react-native';

import Contact from './Contact';

describe('Contact', () => {
  const mockRoute = {
    params: {
      contact: {
        givenName: 'Praveenkumar',
        familyName: 'K',
        phoneNumbers: [
          {
            label: 'mobile',
            number: '+919876543210',
          },
        ],
      },
    },
  };

  it('should match snapshot', () => {
    const container = render(<Contact route={mockRoute} />);

    expect(container).toMatchSnapshot();
  });

  it('should render contact details when route params are present', () => {
    const { getByTestId } = render(<Contact route={mockRoute} />);

    expect(getByTestId('contact_name').props.children).toStrictEqual(
      'Praveenkumar K',
    );
    expect(getByTestId('contact_phone_0').props.children).toStrictEqual(
      '+919876543210',
    );
  });
});
