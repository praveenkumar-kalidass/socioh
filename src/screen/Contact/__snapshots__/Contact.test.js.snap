// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <RNSVGSvgView
    bbHeight={1334}
    bbWidth={750}
    focusable={false}
    height={1334}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 1334,
          "width": 750,
        },
      ]
    }
    width={750}
  >
    <RNSVGGroup>
      <RNSVGCircle
        cx={666.25}
        cy={333.5}
        fill={4294892136}
        propList={
          Array [
            "fill",
          ]
        }
        r={125}
      />
      <RNSVGCircle
        cx={83.75}
        cy={762.2857142857143}
        fill={4279943103}
        propList={
          Array [
            "fill",
          ]
        }
        r={93.75}
      />
      <RNSVGCircle
        cx={650}
        cy={1190.6000000000001}
        fill={4279943103}
        propList={
          Array [
            "fill",
          ]
        }
        r={75}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
